gnpy/core/request.py

```diff
@@ -157,19 +157,19 @@ def pathresult(self):
                    'path-properties':{
                        'path-metric': [
                            {
-                           'metric-type': 'SNR@bandwidth',
```

Comment:
_ am refering to the OpenDayLight yang tool used for transportPCE project. https://docs.opendaylight.org/en/stable-nitrogen/developer-guide/yang-tools.html#working-with-yang-model
I did not catch the error message, but you can see in https://wiki.opendaylight.org/view/YANG_Tools:YANG_to_Java_Mapping that names are converted 

> Some YANG elements are mapped to JAVA classes and interfaces. The name of YANG element may contain various characters which aren't permitted in JAVA class names. Firstly whitespaces are trimmed from YANG name. Next characters space, -, _ are deleted and subsequent letter is capitalized. At the end first letter is capitalized. Transformation example:
example-name without_capitalization
is mapped to
ExampleNameWithoutCapitalization 

since '@' character in java is a reserved key character for decorator, this may explain why the yangtool didn't work with 'SNR@bandwidth' names.

I don't have error message to show right now.

I change the commit description to match the changes

Reply:
Thanks. So as you're explaining in commit e646340b7fed41d1cb9a654d163e0e64564f89a8, this is a workaround to an opendaylight-specific tool. I can live with carrying workarounds, but can you please open a bugreport upstream and ask them to treat ampersands in a similar manner to how they treat, say, dashes or underscores?